### ISSUE TITLE
Browser Authentication (Feathers Guide)

### ISSUE THREAD

**Issue by finch2kt** [feathersjs/docs#1356]
### Comment/Problem
1:Establish socket.io connection
I'm following the feathers guide and I hit a wall at the browser authentication section, [https://docs.feathersjs.com/guides/basics/authentication.html#browser-authentication](url).

I have followed the directions and code edits, to the best of my knowledge, correctly. I noticed that when i first generated within the file src/app.js it defined socketio with `const socketio = require('@feathersjs/socketio');`, but does not define io() in the begnning of the app.js file.

So the problem I'm having/had is that when I added the code for the login method using app.Authenticate(). My IDE tells me that io() is not defined. I did a little digging through the docs and that I should be defining socketio with `const socketio = require('@feathersjs/socketio-client');` and io with `const io = require('socket.io-client');`

And maybe adding a comment within in the code provided for Establishing the socket.io connection to let first time developers know that if they should add their localhost url for the feathers web app started using the guide. Ex const socket = io(<Your localhost url>); 

I'm not sure if it will work yet as I have a second problem with using local storage to store login tokens, as explained below. 

2: Using feathers-localstorage to store login tokens
`// Use localStorage to store our login token
client.configure(feathers.authentication({
  storage: window.localStorage
}));`

My IDE tells me that window is not defined. Should I add feathers-localstorage as a service or just make a variable within the app.js. The part that sticks me is how to incorporate it in the code block above so that it recognizes window through the localstorage module, because I have not found anything definitive about window within my digging through the docs for feathers-localstorage. Everything has told me to initialize a new service like so

`const service = require('feathers-localstorage');

app.use('/messages', service({
  storage: window.localStorage || AsyncStorage
}));
app.use('/messages', service({ storage, id, startId, name, store, paginate }));`

This confuses me, because I know I did not initalize authentication as a service. The example code block above is uses 'feathers-localstorage' with the messages service. so I'm confused on how to properly solve this.

I hope the solving of this or directions on how to go about doing so can help strengthen the feathers guide for other beginners of feathers.js as well as my own understanding of this impressive framework. My teacher really likes it and chose it for our class to use to practice group development processes of a progressive web app, thank you for your work developing this.

Let me know if you need more information or pictures of the orginial errors. I hope I have written this 

### System configuration
**Module versions** (especially the part that's not working): "@feathersjs/socketio@^4.3.0", "@feathersjs/feathers@^4.3.0", and "feathers-localstorage@5.0.0" 

**NodeJS version**: v10.16.3

**Operating System**: Windows 10

**Browser Version**: Version 77.0.3865.75 (Official Build) * Will not successful launch web app so I don't believe it is chrome. Just my initial thoughts, I'm not claiming to be an expert.

**React Native Version**:  Should be the latest as I installed feathers within the last two weeks.

**Module Loader**: Not sure

**Comment by bartduisters** [comment]
1. The [Feathers Client documentation](https://docs.feathersjs.com/api/client.html#node) does have it included.
2. It is not because your IDE claims that it is undefined, that in reality it does not exist. If you use any kind of linting you can use that to ignore this specific error.

**Comment by finch2kt** [comment]
I believe I fixed my issue with getting my app.js code get past establishing the socket.io when I use `npm start`. But I keep getting this error:

C:\Users\kasha\dev\feathers-basics\feathers-chat\node_modules\@feathersjs\a
                throw new Error('${this.name}' authentication strategy req
                ^
Error: 'local' authentication strategy requires a 'usernameField' setting
    at forEach.prop (C:\Users\kasha\dev\feathers-basics\feathers-chat\node_.js:26:23)
    at Array.forEach (<anonymous>)
    at LocalStrategy.verifyConfiguration (C:\Users\kasha\dev\feathers-basicon-local\lib\strategy.js:24:44)
    at AuthenticationService.register (C:\Users\kasha\dev\feathers-basics\flib\core.js:75:22)

I checked my authentication configs in the default.json file, snippet is shown below:
`"authentication": {
    "entity": "user",
    "service": "users",
    "secret": "LnLonAT6nRhxVjsD/z1LgvTe8O4=",
    "authStrategies": ["jwt","local","oauth"],
    "jwtOptions": {
      "header": {
        "typ": "access"
      },
      "audience": "https://yourdomain.com",
      "issuer": "feathers",
      "algorithm": "HS256",
      "expiresIn": "1d"
    },
    "local": {
      "usernameField": "\\email",
      "passwordField": "password"
    },
    "oauth": {
      "redirect": "/",
      "github": {
        "key": "<Client ID>",
        "secret": "<Client Secret>"
      }
    }  
  },`

It looked like I do in fact have local configured with the required username name setting. Only thing I don't understand is how the entity, user, and the service, users, may effect the local authentication strategy. I also tried a tip from someone having a problems with authentication-local where you use "\\email" in the default.json to escape the configs and therefore feathers does not look them up in the environment from #294 . This still did not work

I followed the errors being thrown start with this snippet of code from src/app.js, `// Use localStorage to store our login token
client.configure(authentication(client,{
  // eslint-disable-next-line no-undef
  storage: localStorage
}
));`, then from src/authentication.js the authentication service calls the register method which then calls LocalStrategy.verify.Configuration and it is here that the code says that the prop usernameField does not equal a string so it throws the error that Local authentication needs usernameField as a setting.

Since I'm still new to featherjs, the only other thing I can think of would be the user.hooks.js file, because in the before clause the "local" authentication-local local strategy is not referenced.

I hope I explained my issue properly for whoever responds to this.
